src/main/java/com/pp/productsapp/ProductController.java

```diff
@@ -0,0 +1,56 @@
+package com.pp.productsapp;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+
+@Controller
+public class ProductController {
+
+    private ProductRepository productRepository;
+
+    public ProductController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @GetMapping("/index")
+    public String index(Model model) {
+        model.addAttribute("product", new Product());
+        return "index";
+    }
+
+    @GetMapping("/list")
+    public Object list(@RequestParam(required = false) String category, Model model) {
+
+        if (category == null || category.equals("")) {
+            int sumPrice = 0;
+            List<Product> productList = productRepository.getAll();
+            sumPrice = productRepository.getPrice(category);
```

Comment:
Nie przekazuj kategorii do repozytorium w celu policzenia ceny. Masz już w tym miejscu listę produktów, dla którcyh powinieneś zsumować cenę, zrób to tutaj w pętli/streamie ;)